url-dsl/src/main/scala/urldsl/language/PathSegmentImpl.scala

```diff
@@ -28,8 +28,9 @@ trait PathSegmentImpl[A] {
   /** implementation of [[urldsl.errors.PathMatchingError]] for type A. */
   implicit protected val pathError: PathMatchingError[A]
 
-  val root: PathSegment[Unit, A] = PathSegment.root
-  val remainingSegments: PathSegment[List[String], A] = PathSegment.remainingSegments
+  lazy val root: PathSegment[Unit, A] = PathSegment.root
+  lazy val remainingSegments: PathSegment[List[String], A] = PathSegment.remainingSegments
+  lazy val ignoreRemaining: PathSegment[Unit, A] = remainingSegments.ignore(Nil)
```

Comment:
Naming comment: hmm `ignoreRemaining` seems unclear on its own – remaining what? Especially when used as a property, without the `/`. Maybe `ignoreRemainingSegments`? – even if it's a mouthful, it would be clearer and consistent. Usages of it should be pretty rare anyway...

Reply:
I chose `ignoreRemaining` over `ignoreRemainingSegments` because I thought it was too long, but that's indeed not a good point. It's better to be more explicit.